tests/Parallafka.Tests/CommitStateTests.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Parallafka.KafkaConsumer;
+using Xunit;
+
+namespace Parallafka.Tests
+{
+    public class CommitStateTests
+    {
+        [Fact]
+        public async Task EnqueuesOnlyUpToMax()
+        {
+            //  given
+            var stop = new CancellationTokenSource();
+            var cs = new CommitState<string, string>(5, stop.Token);
+            var messages = Enumerable.Range(0, 6)
+                .Select(i => new KafkaMessage<string, string>("key", "value", new RecordOffset(0, i)))
+                .ToList();
+
+            foreach (var message in messages.Take(5))
+            {
+                await cs.EnqueueMessageAsync(message);
+            }
+
+            var message6 = messages.Last();
+            var waitToken = new CancellationTokenSource(TimeSpan.FromSeconds(1));
+
+            // when/then
+            var enqueueTask = cs.EnqueueMessageAsync(message6);
+            await Assert.ThrowsAnyAsync<OperationCanceledException>(async () =>
+            {
+                var task = await Task.WhenAny(enqueueTask, Task.Delay(-1, waitToken.Token));
+                await task;
+            });
+
+            // when
+            var message1 = messages.First();
+            message1.WasHandled = true;
+
+            // then
+            cs.GetMessagesToCommit().ToList();
+            await enqueueTask;
```

Comment:
If I'm understanding this - ideally this would be awaited with `Wait.ForTaskOrTimeout` so broken code doesn't hang the test